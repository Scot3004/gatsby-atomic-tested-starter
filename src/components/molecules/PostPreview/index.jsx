import React from 'react'
import Card from '../../atoms/Card'
import CardContent from '../../atoms/CardContent'
import Avatar from '../../atoms/Avatar'
import PropTypes from 'prop-types'
import Excerpt from '../../atoms/Excerpt'
import { Title } from '../../atoms/Title'

export const PostPreview = props => (
  <Card>
    <Avatar src={props.image} alt={props.title} size='64px' />
    <CardContent>
      <Title>{props.title}</Title>
      <Excerpt>{props.excerpt}</Excerpt>
    </CardContent>
  </Card>
)

PostPreview.propTypes = {
  image: PropTypes.string,
  title: PropTypes.string,
  excerpt: PropTypes.string
}

export default PostPreview
