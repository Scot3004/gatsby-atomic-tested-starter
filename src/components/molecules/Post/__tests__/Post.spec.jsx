import React from 'react'
import 'jest-styled-components'
import renderer from 'react-test-renderer'
import PostPreview from '..'

const stub = {
  title: 'Title stub',
  image: 'image_stub',
  frontmatter: {
    date: 'date stub'
  },
  html: 'html stub'
}

test('it works', () => {
  const tree = renderer.create(<PostPreview {...stub} />).toJSON()
  expect(tree).toMatchSnapshot()
})
