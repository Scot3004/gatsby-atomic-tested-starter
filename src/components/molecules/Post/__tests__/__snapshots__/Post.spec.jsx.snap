// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it works 1`] = `
.c1 {
  -webkit-flex: 0 0 64px;
  -ms-flex: 0 0 64px;
  flex: 0 0 64px;
  width: 64px;
  height: 64px;
  padding: 10px;
}

.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-shadow: 0 4px 8px 0 rgba(0,0,0,0.2);
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-bottom: 15px;
  -webkit-transition: 0.3s;
  transition: 0.3s;
  width: 100%;
  padding: 10px;
}

.c0:hover {
  box-shadow: 0 6px 12px 0 rgba(0,0,0,0.2);
}

.c0:last-child {
  margin-bottom: 0;
}

.c2 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

<div
  className="c0"
>
  <img
    alt="Title stub"
    className="c1"
    size="64px"
    src="image_stub"
  />
  <div
    className="c2"
  >
    <small>
      date stub
    </small>
    <div
      className="blog-post-content"
      dangerouslySetInnerHTML={
        Object {
          "__html": "html stub",
        }
      }
    />
  </div>
</div>
`;
